SignalMessaging/environment/OWSSounds.m

```diff
@@ -0,0 +1,463 @@
+        case OWSSound_Crystals:
+            return @"Crystals.m4r";
+        case OWSSound_Hillside:
+            return @"Hillside.m4r";
+        case OWSSound_Illuminate:
+            return @"Illuminate.m4r";
+        case OWSSound_Night_Owl:
+            return @"Night Owl.m4r";
+        case OWSSound_Opening:
+            return @"Opening.m4r";
+        case OWSSound_Playtime:
+            return @"Playtime.m4r";
+        case OWSSound_Presto:
+            return @"Presto.m4r";
+        case OWSSound_Radar:
+            return @"Radar.m4r";
+        case OWSSound_Radiate:
+            return @"Radiate.m4r";
+        case OWSSound_Ripples:
+            return @"Ripples.m4r";
+        case OWSSound_Sencha:
+            return @"Sencha.m4r";
+        case OWSSound_Signal:
+            return @"Signal.m4r";
+        case OWSSound_Silk:
+            return @"Silk.m4r";
+        case OWSSound_Slow_Rise:
+            return @"Slow Rise.m4r";
+        case OWSSound_Stargaze:
+            return @"Stargaze.m4r";
+        case OWSSound_Summit:
+            return @"Summit.m4r";
+        case OWSSound_Twinkle:
+            return @"Twinkle.m4r";
+        case OWSSound_Uplift:
+            return @"Uplift.m4r";
+        case OWSSound_Waves:
+            return @"Waves.m4r";
+
+            // Calls
+        case OWSSound_CallConnecting:
+            return @"ringback_tone_cept.caf";
+        case OWSSound_CallOutboundRinging:
+            return @"ringback_tone_ansi.caf";
+        case OWSSound_CallBusy:
+            return @"busy_tone_ansi.caf";
+        case OWSSound_CallFailure:
+            return @"end_call_tone_cept.caf";
+    }
+}
+
++ (NSURL *)soundURLForSound:(OWSSound)sound
+{
+    NSString *filename = [self filenameForSound:sound];
+    NSURL *_Nullable url = [[NSBundle mainBundle] URLForResource:filename.stringByDeletingPathExtension
+                                                   withExtension:filename.pathExtension];
+    OWSAssert(url);
+    return url;
+}
+
++ (void)playSound:(OWSSound)sound
+{
+    [self.sharedManager playSound:sound];
+}
+
+- (SystemSoundID)systemSoundIDForSound:(OWSSound)sound
+{
+    @synchronized(self)
+    {
+        NSNumber *_Nullable systemSoundID = self.systemSoundIDMap[@(sound)];
+        if (!systemSoundID) {
+            NSURL *soundURL = [OWSSounds soundURLForSound:sound];
+            SystemSoundID newSystemSoundID;
+            OSStatus error = AudioServicesCreateSystemSoundID((__bridge CFURLRef)soundURL, &newSystemSoundID);
+            if (error) {
+                OWSFail(@"%@ could not load sound.", self.logTag);
+            }
+            systemSoundID = @(newSystemSoundID);
+            self.systemSoundIDMap[@(sound)] = systemSoundID;
+        }
+        return (SystemSoundID)systemSoundID.unsignedIntegerValue;
+    }
+}
+
+- (void)playSound:(OWSSound)sound
+{
+    SystemSoundID systemSoundID = [self systemSoundIDForSound:sound];
+    AudioServicesPlayAlertSound(systemSoundID);
```

Comment:
We might want to use AVAudioPlayer instead...

Reply:
Do you know how this plays with background audio? e.g. I know if we're using AVAudioPlayer we'll have to explicitly end the audio session afterwards. Not sure about AudioServices*

Reply:
I've made a item in the checklist for the card.